pages/shields/led_matrix.rst

```diff
@@ -1,4 +1,190 @@
+terminal software of choice). Now let's run the following commands to get
+these LEDs lit::
+
+    import machine
+    import wemos
+    import tm1640
+    tm = tm1640.TM1640(clk=machine.Pin(wemos.D5), dio=machine.Pin(wemos.D7))
+    tm.write([255,255,255,255,255,255,255,255])  # 255 = 0b11111111
+
+Now all LEDs on the LED Matrix should be illuminated! Now lets run through what
+we just did:
+
+- Imported the MicroPython ``machine`` module -- we need this to configure our
+  pins before passing them to our ``tm1640`` driver
+- Imported our ``wemos`` module -- we need this to get the pin information to
+  then configure the correct pins for communicating with the LED Matrix driver
+  chip
+- Imported the ``tm1640`` module -- this is the MicroPython driver that will
+  provide us easy control over the ``tm1640`` LED matrix driver chip on the
+  shield
+- Created a ``tm`` object that represents our LED Matrix driver. This took two
+  parameters, the two pins that are used for communicating with the ``tm1640``
+  chip (a clock pin and a data in/out pin)
+- Wrote an 8-long list filled with 255's to our ``tm`` object -- this is what
+  turns the LED Matrix LEDs on!
+  
+The format of the list that we wrote to the ``tm`` object is that each element
+of the list represents a row, and each bit in that element represents one of
+the LEDs in that row, with 1 being illuminated and 0 being off. As
+``255 == 0b11111111``, and we wrote 255 to all 8 elements of the list, we
+illuminated all 8 LEDs, on all 8 rows.
+
+Our driver library also gives us access to a ``brightness()`` method on our
+``tm`` object, if we wanted to reduce the brightness of the LED matrix. This
+takes a single argument of an integer from 0 to 7, where 0 is the minimum
+brightness, and 7 is the maximum (this is the default). So if we wanted to
+reduce brightness by a bit but not all the way we could do::
+
+    tm.brightness(3)
+
+If you wanted to (slightly) improve visualisation of what you are writing to
+the matrix in your code, you could format it like so::
+
+    tm.write([
+        0b00000000,
+        0b01100110,
+        0b01100110,
+        0b00000000,
+        0b00000000,
+        0b10000001,
+        0b01111110,
+        0b00000000,
+    ])
+
+Blue pill, red pill
+===================
+
+Now that you know how to light up LEDs individually, it's time to learn
+about some convenience functions that can help display *text*.
+
+Let's display a letter on the matrix::
+
+    tm1640.display_letter(tm, "X")
+
+And, for the pièce de résistance::
+
+    tm1640.scroll_text(tm, "Scrolling for days...")
+
+These are enabled by using a *FrameBuffer*, a module built-in to MicroPython
+that provides a general - and efficient! - way to draw onto an in-memory
+'canvas'.
+
+Advanced: FrameBuffer
+=====================
+
+A flexible way to control the LEDs in the matrix is by using a
+`MicroPython frame buffer`_. This is done like
```

Comment:
Maybe should mentioned that this is the same FrameBuffer talked about in previous section